datahub-web-react/src/app/entityV2/shared/tabs/Incident/utils.tsx

```diff
@@ -387,7 +393,9 @@ const getFilteredIncidents = (incidents: Incident[], filter: IncidentListFilter)
 
     // Apply type, priority, and stage
     return incidents.filter((incident: Incident) => {
-        const matchesCategory = type.length === 0 || type.includes(incident.incidentType);
+        const category =
+            incident.incidentType === IncidentType.Custom ? incident.customType ?? '' : incident.incidentType ?? '';
```

Comment:
would prefer to remove any ?? '' in the PR if possible (without casting to string)